package-scripts/datadog-agent/prerm

```diff
@@ -28,8 +28,11 @@ else
     exit 1;
 fi
 
-# Delete all.pyc files
+# Delete all.pyc files in the agent's dir
 find /opt/datadog-agent/agent -name '*.py[co]' -type f -delete || echo 'Unable to delete .pyc files'
 
-exit 0
+# Delete all the .pyc files in the embedded dir that are part of the agent's package
+# (i.e. not from packages installed manually by the user)
+cat /opt/datadog-agent/embedded/py_compiled_files.txt | xargs rm -f
```

Comment:
Actually the permissions are `644 root:root`, we could restrict it more just to make it clear that the file is not meant to be modified.

And you're right currently the command would fail if the file isn't there. I can test that the file is there beforehand.